### PR TITLE
Fix some failed test cases

### PR DESCRIPTION
**Pull request by hoanhtien**
BasicIntellisense/CtrlSpace
BasicProjectTests/AddFolderExists
BasicProjectTests/TestAddExistingFolderProject
BasicProjectTests/CopyFullPath

### CODE REVIEW

File: Nodejs/Tests/Core.UI/BasicProjectTests.cs [hunk 1]
```diff
@@ -993,6 +993,20 @@ public class BasicProjectTests : SharedProjectTest {
...
+        /// </summary>
+        private static string removeLabel(string itemName) {
+            string[] labels = new string[3] { " (browser)", " (node)", " (node, browser)" };
+            foreach (var label in labels) {
+                if (itemName.EndsWith(label)) {
+                    itemName = itemName.Substring(0, itemName.Length - label.Length);
+                    break; // safe to break as there is at most one label per name
+                }
+            }
+            return itemName;
```
Comment (mousetraps) [inline Nodejs/Tests/Core.UI/BasicProjectTests.cs]: Instead of messing around with the string (which is going to break if we ever change the string or the option is turned off), it's better to just turn off the option before running the test. e.g. 
https://github.com/Microsoft/nodejstools/blob/8169f1e6fe7c83e3f4061631a44de79b3e80ec73/Nodejs/Tests/Core.UI/NodejsBasicProjectTests.cs#L272

File: Nodejs/Tests/Core.UI/BasicProjectTests.cs [hunk 2]
```diff
@@ -722,7 +722,7 @@ public class BasicProjectTests : SharedProjectTest {
                     Keyboard.Type(System.Windows.Input.Key.Enter);
 
 #if DEV11_OR_LATER
-                    VisualStudioApp.CheckMessageBox(MessageBoxButton.Ok, "Directory names cannot contain any of the following characters");
+                    VisualStudioApp.CheckMessageBox(MessageBoxButton.Ok, "Please enter a valid name.");
```
Comment (mousetraps) [inline Nodejs/Tests/Core.UI/BasicProjectTests.cs]: Have you tested this across VS versions?

File: Nodejs/Tests/Core.UI/BasicIntellisense.cs [hunk 3]
```diff
@@ -45,6 +45,7 @@ public sealed class BasicIntellisense : NodejsProjectTest {
 
                 server.MoveCaret(2, 13);
                 solution.ExecuteCommand("Edit.CompleteWord");
+                Keyboard.Type("\t");
```
Comment (mousetraps) [inline Nodejs/Tests/Core.UI/BasicIntellisense.cs]: why is this necessary? Edit.CompleteWord should be sufficient, and we should try to fix the ExecuteCommand call if that's not working.

Comment (hoanhtien) [inline Nodejs/Tests/Core.UI/BasicIntellisense.cs]: Edit.CompleteWord is not doing what we think it is. Looking at [IntellisenseControler.cs#L646](https://github.com/Microsoft/nodejstools/blob/5c994768c5678579b76d71ba6ff48a30fa5991cd/Nodejs/Product/Nodejs/Intellisense/IntellisenseController.cs#L646), it does the same thing with ShowMemberList.

File: Nodejs/Tests/Core.UI/BasicProjectTests.cs (hunk 1 above)
Comment (hoanhtien) [inline Nodejs/Tests/Core.UI/BasicProjectTests.cs]: After a second look, I will remove this change as it is not related to the three test cases I am fixing in this pull request (the only test case that is affected by this change is BasicProjectTests\LoadProjectWithDuplicateItems and it still fails after this change).

File: Nodejs/Tests/Core.UI/BasicProjectTests.cs (hunk 2 above)
Comment (hoanhtien) [inline Nodejs/Tests/Core.UI/BasicProjectTests.cs]: Looks like it does not work on VS2013 :(

**Comment by hoanhtien** [comment]
Updated the pull request to address feedback.

**Comment by mjbvz** [comment]
Resubmitted and updated PR as #827
